pkgs/applications/emulators/ukmm/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, atk
+, glib
+, cmake
+, gtk3-x11
+, pkg-config
+, rustPlatform
+, wrapGAppsHook
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ukmm";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "NiceneNerd";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-TvRaaAVJPqvTTqs+d5d4y1KmDTvE41uP5AykLtHxD9w=";
+  };
+
+  cargoLock = {
+      lockFile = ./Cargo.lock;
+      outputHashes = {
+        "catppuccin-egui-1.0.2" = "sha256-+ILfvDgZxe/QPJuVqIbRjaHNovpRAX+ym2QZ96glb4w=";
+        "ecolor-0.20.0" = "sha256-uTDkNRWsA1nM8Qhb0X2LjVDRuaW31vWxR8kDLL27BVE=";
+        "egui-notify-0.4.0" = "sha256-jybtUnv9xqzulZ5nfg+T1u8iTOsPjKGVVQ7JhwbvPdU=";
+        "egui_commonmark-0.6.0" = "sha256-hsVbtL2F+jifnzN6FgcDAVtLd1bVxTs0twn0SMvq9eU=";
+        "egui_dock-0.2.1" = "sha256-gGIO0boXKxLu0ABDH/uJhEZEoE/ql8E65LRmr0Xhv3s=";
+        "junction-0.2.0" = "sha256-6+pPp5wG1NoIj16Z+OvO4Pvy0jnQibn/A9cTaHAEVq4=";
+        "msbt-0.1.1" = "sha256-PtBs60xgYrwS7yPnRzXpExwYUD3azIaqObRnnJEL5dE=";
+        "msyt-1.2.1" = "sha256-aw5whCoQBhO0u9Fx2rTO1sRuPdGnAAlmPWv5q8CbQcI=";
+      };
+  };
+
+  doCheck = false;
```

Comment:
I'm also curious - what was wrong with the tests?

Reply:
```
...
...
...
     Compiling ukmm v0.7.1 (/build/source)
      Finished release [optimized] target(s) in 1m 21s
       Running unittests src/main.rs (target/x86_64-unknown-linux-gnu/release/deps/ukmm-681b477a3355b37b)
   
  running 1 test
  test gui::tasks::tests::remerge ... FAILED
   
  failures:
   
  ---- gui::tasks::tests::remerge stdout ----
  thread 'gui::tasks::tests::remerge' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to initialize mod manager at `crates/uk-manager/src/core.rs@21:44`
   
  Caused by:
      0: failed to create directory `/homeless-shelter/.local/share/ukmm/wiiu/profiles/Default`
      1: Permission denied (os error 13)', src/gui/tasks.rs:541:54
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
   
   
  failures:
      gui::tasks::tests::remerge
   
  test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
   
  error: test failed, to rerun pass `--bin ukmm`
```

Reply:
Oh, I think you can fix that with:

```nix
preCheck = ''
  export HOME=$TMDIR
'';
```

Reply:
```diff
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';```
```
```
     Compiling ukmm v0.7.1 (/build/source)
      Finished release [optimized] target(s) in 1m 15s
       Running unittests src/main.rs (target/x86_64-unknown-linux-gnu/release/deps/ukmm-681b477a3355b37b)
   
  running 1 test
  test gui::tasks::tests::remerge ... FAILED
   
  failures:
   
  ---- gui::tasks::tests::remerge stdout ----
  thread 'gui::tasks::tests::remerge' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to apply pending mod changes
   
  Caused by:
      No dump available for current platform at `crates/uk-manager/src/deploy.rs@443:14`', src/gui/tasks.rs:542:51
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
   
   
  failures:
      gui::tasks::tests::remerge
   
  test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
   
  error: test failed, to rerun pass `--bin ukmm`
```